tarpc/tests/service_functional.rs

```diff
@@ -136,6 +136,45 @@ async fn serde() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[cfg(all(feature = "serde-transport", feature = "unix", unix))]
+#[tokio::test]
+async fn serde_uds() -> anyhow::Result<()> {
+    use std::fs;
+    use tarpc::serde_transport;
+    use tokio_serde::formats::Json;
+
+    const SOCK: &str = "tarpc_serde.sock";
+
+    // clean any existing temporary sockets that happened to exist
+    let _ = fs::remove_file(SOCK);
+
+    let _ = tracing_subscriber::fmt::try_init();
+
+    let transport = tarpc::serde_transport::unix::listen(SOCK, Json::default).await?;
+    tokio::spawn(
+        transport
+            .take(1)
+            .filter_map(|r| async { r.ok() })
+            .map(BaseChannel::with_defaults)
+            .execute(Server.serve()),
+    );
+
+    let transport = serde_transport::unix::connect(SOCK, Json::default).await?;
+    let client = ServiceClient::new(client::Config::default(), transport).spawn();
+
+    // Save results using socket so we can clean the socket even if our test assertions fail
+    let res1 = client.add(context::current(), 1, 2).await;
+    let res2 = client.hey(context::current(), "Tim".to_string()).await;
+
+    // Clean temporary socket even if assertions fail
+    let _ = fs::remove_file(SOCK);
```

Comment:
optional nit: Another way to accomplish this would be with a cleanup struct initialized up above, when the socket path is initialized. That way, the cleanup code is a bit out of the way of the main test code.

```suggestion
    // Clean temporary socket even if assertions fail
   struct Cleanup(std::path::PathBuf);
   impl Drop for Cleanup {
       fn drop(&mut self) {
           let _ = fs::remove_file(self.0);
       }
   }
```

Reply:
Because the unix sockets have some other requirements like implementing `AsRef<Path>` I went ahead and added a `TempSock` to the `unix` module. If you'd prefer, I can remove that and just duplicate the code in both tests, but it seems like a useful thing to me even for non-test code.

`TempSock`: [0d6fbdb](https://github.com/google/tarpc/pull/380/commits/0d6fbdbec2f8e9213138ce4a26bb132e7e9a2a3e)
Test cleanup: [ce5715c](https://github.com/google/tarpc/pull/380/commits/ce5715c93f0e58eef45bf0f2448b3a0d7f5fda11)